internal/controller/controllers/bmo_utils.go

```diff
@@ -59,6 +60,10 @@ func NewBMOUtils(client client.Reader, osClient *osclientset.Clientset, kubeClie
 
 // ConvergedFlowAvailable checks the baremetal operator version and returns true if it's equal or higher than the minimal version for converged flow
 func (r *bmoUtils) ConvergedFlowAvailable() bool {
+	if !r.imageServiceEnabled {
```

Comment:
Why are these things related?

Reply:
With converged flow we setup the ISO directly in preprovisioning image. As the application will be unaware of the base image, I don't think it would make much sense using the converged flow. Would you agree?

Reply:
But without converged flow we set the ISO directly in the BMH.
We shouldn't be doing either so I don't think switching between them based on the image service status really makes much difference so I'd rather not make this change.

Reply:
We want to guard against (or sanely report to the user) the situation where someone creates a BMH with the infraenv label because we know this won't work without the image service.